ArduCopter/ekf_check.cpp

```diff
@@ -121,11 +129,8 @@ bool Copter::ekf_over_threshold()
     } else if (vel_variance >= g.fs_ekf_thresh) {
         over_thresh_count++;
     }
-    if (position_variance >= g.fs_ekf_thresh) {
-        over_thresh_count++;
-    }
 
-    if (over_thresh_count >= 2) {
+    if (position_variance >= g.fs_ekf_thresh && over_thresh_count >= 1) {
```

Comment:
this seems to preclude the possibility of an EKF failsafe due to large height innovations (eg. z vibe). Do we really want to preclude that?

Reply:
The changes  don't preclude preclude it because height isn't being checked in the original code. The way copter uses the output from ekf_over_threshold makes it fairly pointless to check height as the fallback modes use the height estimate and this logic is primarily there for loss of horizontal control caused by GPS glitches and bad yaw.

I will modify the check to 

(position_variance >= g.fs_ekf_thresh && over_thresh_count >= 1) || over_thresh_count >= 2

so that the slower response condition  with two or more criteria is preserved.

Reply:
@rmackay9 As part of a separate activity I would like to revisit the EKF failsafe in light of the increased functionality and fallback options of our estimators.  At the moment we have separate ekf_check, check_vibration and gpsglitch_check tasks which run one after the other at the same 10Hz rate.